backend/canisters/user/api/can.did

```diff
@@ -117,6 +117,28 @@ type RemoveReactionResponse = variant {
     UserSuspended;
 };
 
+type TipMessageArgs = record {
+    chat : Chat;
+    recipient : UserId;
```

Comment:
Why do we need the recipient when we have the messageId?

Reply:
We use the recipient to validate the transfer is going to the right place, we don't actually need it for anything else so we could remove it, I just left it in their as a safety net in case the FE every passes up the wrong transfer details.

Reply:
Actually we can extract the userId from the transfer and then validate that matches the message sender, so we can remove it.

Reply:
But we are tipping the _message_ - surely the recipient comes from the message and if we verify anything it's that the transfer matches the sender of the message.

Reply:
Yeah we verify that the recipient matches the message sender, but the message sender isn't known by the user canister in the case of a group or channel message so I was taking the recipient arg in order to verify the transfer is correct before we make the transfer.

Reply:
But having the recipient doesn't really make things any better as the FE could pass up the incorrect recipient along with a matching incorrect transfer, the user canister would think everything is fine and so make the transfer and forward the tip on the group, and only then would the sender validation work correctly.
So I'll remove the recipient, extract the user_id from the transfer, then only add the tip if the user_id matches the message sender.
That way if someone hacks the FE they could cause the payment to be made, but no tip will actually display unless the transfer is to the message's sender.